packages/sanity/src/core/store/_legacy/document/document-pair/remoteSnapshots.ts

```diff
@@ -1,16 +1,17 @@
-import {SanityClient} from '@sanity/client'
-import {merge} from 'rxjs'
+import type {SanityClient} from '@sanity/client'
+import {merge, type Observable} from 'rxjs'
 import {switchMap} from 'rxjs/operators'
-import {IdPair} from '../types'
-import {memoize} from '../utils/createMemoizer'
+import type {IdPair} from '../types'
+import type {RemoteSnapshotVersionEvent} from './checkoutPair'
 import {memoizedPair} from './memoizedPair'
 
 /** @internal */
-export const remoteSnapshots = memoize(
-  (client: SanityClient, idPair: IdPair, typeName: string) => {
-    return memoizedPair(client, idPair, typeName).pipe(
-      switchMap(({published, draft}) => merge(published.remoteSnapshot$, draft.remoteSnapshot$))
-    )
-  },
-  (_client, idPair, typeName) => idPair.publishedId + typeName
-)
+export const remoteSnapshots = (
```

Comment:
We still need the memoize wrapper here, and it should use the same key generator as memoizedPair. 

Although removing memoization here will still benefit from using the memoized observable returned from `memoizedPair`, it will now make `remoteSnapshot` return a new observable instance every time it's called, which might break some expectations down the line.